modules/data.atmosphere/R/extract_local_CMIP5.R

```diff
@@ -0,0 +1,299 @@
+##' @param ensemble_member which CMIP5 experiment ensemble member
+##' @param overwrite logical. Download a fresh version even if a local file with the same name already exists?
+##' @param verbose logical. to control printing of debug info
+##' @param ... Other arguments, currently ignored
+##' @export
+##' @examples
+# -----------------------------------
+extract.local.CMIP5 <- function(outfolder, in.path, start_date, end_date, site_id, lat.in, lon.in, 
+                                model , scenario , ensemble_member = "r1i1p1",
+                                overwrite = FALSE, verbose = FALSE, ...){
+  library(lubridate)
+  library(ncdf4)
+  library(stringr)
+  
+  # Some GCMs don't do leap year; we'll have to deal with this separately
+  no.leap <- c("bcc-csm1-1", "CCSM4")
+  
+  # Days per month
+  dpm <- days_in_month(1:12)
+  
+  # Date stuff
+  start_date <- as.POSIXlt(start_date, tz = "GMT")
+  end_date <- as.POSIXlt(end_date, tz = "GMT")
+  start_year <- year(start_date)
+  end_year   <- year(end_date)
+  
+  lat.in = as.numeric(lat.in)
+  lon.in = as.numeric(lon.in)
+  # dir.nldas="http://hydro1.sci.gsfc.nasa.gov/thredds/dodsC/NLDAS_FORA0125_H.002"
+  dir.create(outfolder, showWarnings=FALSE, recursive=TRUE)
+  
+  ylist <- seq(start_year,end_year,by=1)
+  rows = length(ylist)
+  results <- data.frame(file=character(rows), host=character(rows),
+                        mimetype=character(rows), formatname=character(rows),
+                        startdate=character(rows), enddate=character(rows),
+                        dbfile.name = "NLDAS",
+                        stringsAsFactors = FALSE
+                        )
+  
+  # The table of var name conversion
+  # psl; sfcWind; tasmax; tasmin; huss
+  var <- data.frame(DAP.name = c("tas", "tasmax", "tasmin", "rlds", "ps", "rsds", "uas", "vas", "sfcWind", "ua", "va", "huss", "pr"), 
+                    CF.name = c("air_temperature", "air_temperature_maximum", "air_temperature_minimum", 
+                                "surface_downwelling_longwave_flux_in_air",
+                                "air_pressure", "surface_downwelling_shortwave_flux_in_air", 
+                                "eastward_wind", "northward_wind", "wind_speed", "eastward_wind", "northward_wind", 
+                                "specific_humidity", "precipitation_flux"), 
+                    units = c("Kelvin", "Kelvin", "Kelvin", "W/m2", "Pascal", "W/m2", "m/s", "m/s", "m/s", "m/s", "m/s", "g/g", "kg/m2/s"))
+  
+  # Figuring out what we have daily for and what we only have monthly for
+  vars.gcm.day <- dir(file.path(in.path, "day"))
+  vars.gcm.mo <- dir(file.path(in.path, "month"))
+  vars.gcm.mo <- vars.gcm.mo[!vars.gcm.mo %in% vars.gcm.day]
+  
+  vars.gcm <- c(vars.gcm.day, vars.gcm.mo)
+  
+  # Rewriting the dap name to get the closest variable that we have for the GCM (some only give uss stuff at sea level)
+  library(car) # having trouble gettins stuff to work otherwise
```

Comment:
same comment as the library() call above, standard PEcAn convention is using car::function instead of calling the entire library.